raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -129,13 +135,17 @@
     private final ConnectionCache connections;
     private final FuturePurgatory<Void> purgatory;
     private final BlockingQueue<PendingAppendRequest> unsentAppends;
+    private final SortedMap<OffsetAndEpoch, BaseOffsetAndTime> pendingAppends;
 
     private ReplicatedStateMachine stateMachine;
 
+    private long electionStartMs = -1L;
```

Comment:
I tried to push this into the CandidateState but found that it introduced more complexity since we then need to add the metrics into the state or the timer. LMK if you have a better idea.

Reply:
Let me try to push it inside the metrics class with two funcs.